bibtex-completion.el

```diff
@@ -1030,16 +1046,20 @@ line."
 (defun bibtex-completion-show-entry (keys)
   "Show the first selected entry in the BibTeX file."
   (catch 'break
-    (dolist (bibtex-file (-flatten (list bibtex-completion-bibliography)))
+    (dolist (bib-file (bibtex-completion-normalize-bibliography 'main))
       (let ((key (car keys))
-            (buf (bibtex-completion-buffer-visiting bibtex-file)))
-        (find-file bibtex-file)
-        (goto-char (point-min))
-        (if (re-search-forward
-             (concat "^@\\(" parsebib--bibtex-identifier
-                     "\\)[[:space:]]*[\(\{][[:space:]]*"
-                     (regexp-quote key) "[[:space:]]*,") nil t)
-            (throw 'break t)
+            (buf (bibtex-completion-buffer-visiting bib-file)))
+        (find-file bib-file)
+        (if (eq major-mode 'org-mode)
+            (when (ignore-errors (org-link-search (concat "#" key)))
```

Comment:
I confirm that the buffer wasn't widened, so I added `widen`.

Also the function `org-link-search`, when passed a string `"#key"`, basically just calls `(org-find-property "CUSTOM_ID" "key")`. Moreover, org-bibtex has the variable `org-bibtex-key-property`, which defaults to "CUSTOM_ID" but can be set by the user. So I directly used the latter function, as well as the latter variable if bound.

Finally, I realized that `(unless buf (kill-buffer))` at the end of the function was misplaced (it only applied to bibtex files and not to org files).